src/components/editor-content/editor-content.tsx

```diff
@@ -13,17 +16,122 @@ import TimeUtils from "utils/time-utils";
 interface Props {
 }
 
+/**
+ * Values for filtering scopes
+ */
+enum FilterScopes {
+  WEEK = "week",
+  DATE = "date",
+  MONTH = "month",
+  YEAR = "year"
+};
+
 /**
  * Application editor content component
  *
  * @param props component properties
  */
 const EditorContent: React.FC<Props> = () => {
+
   const classes = useEditorContentStyles();
+
   const dispatch = useAppDispatch();
 
   const { person, personTotalTime } = useAppSelector(selectPerson);
 
+  const [selectedStartingDate, setSelectedStartingDate] = useState<Date>(new Date());
+  const [selectedEndingDate, setSelectedEndingDate] = useState<Date>(new Date());
+  const [dateFormat, setDateFormat] = React.useState<string>("dd/M/yyyy");
+  const [scope, setScope] = React.useState<DatePickerView>("date");
+  // TODO: Dynamically check week number and how many weeks each year has
+  const [startWeek, setStartWeek] = React.useState<Number>(1);
+  const [endWeek, setEndWeek] = React.useState<Number>(53);
+
+  /**
+   * Method to handle starting date change
+   *
+   * @param date 
+   */
+  const handleStartDateChange = (date: Date | null) => {
+    if (!date) {
+      return;
+    }
+
+    setSelectedStartingDate(date);
+  };
+
+  /**
+   * Method to handle ending date change
+   * @param date 
+   */
+  const handleEndDateChange = (date: Date | null) => {
+    if (!date) {
+      return;
+    }
+
+    setSelectedEndingDate(date);
+  };
+  
+  /**
+   * Method to handle starting week change
+   * @param date 
+   */
+  const handleStartWeekChange = (event: React.ChangeEvent<{ value: unknown }>) => {
```

Comment:
Change the <Select> components to <TextField> with property select and change type of event to

event: React.ChangeEvent<HTMLInputElement>